File: cs50/pset6/caesar.py
import cs50
import sys

if len(sys.argv) == 2:
    k =  int(sys.argv[1])
    
    s = input()
    ss = ""
    
    for symbol in s:
        
        if symbol.islower():
            ascii = ord(symbol) + k%26
            
            if ascii > 122:
                ascii = ascii - 122 + 96
                
            buf = chr(ascii)
        
        elif symbol.isupper():
            ascii = ord(symbol) + k%26
            
            if ascii > 90:
                ascii = ascii - 90 + 64
                
            buf = chr(ascii)
            
        else:
            buf = symbol
                
        ss += buf

    print(ss)
    
else:
    print("Please enter one argument next time!")

File: cs50/pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[]){
    
    if( argc != 2)
    {
        printf("Please enter one argument next time!\n");
        return 1;
    }
    
    int k = atoi(argv[1]);
    
    string s = GetString();
    
    int p[strlen(s)];
    
    for (int i = 0, len = strlen(s); i < len; i++)
    {
        if( islower(s[i]) )
        {
            
            p[i] = s[i] + k%26;
            
            if(p[i] > 122)
            {
                p[i] = p[i] - 122 + 96;
            }
            
        }
        
        else if( isupper(s[i]) )
        {
            p[i] = s[i] + k%26;
            
            if(p[i] > 90)
            {
                p[i] = p[i] - 90 + 64;
            }
            
        }
        
        else
        {
            p[i] = s[i];
        }
        
        printf("%c", p[i]);
    }
    
    printf("\n");
    
    return 0;
}